### PR TITLE
File format fix

### PR DESCRIPTION
**Pull request by JonathanB-Vobling**
If a GLTF had URL query strings as part of its image URLs it GLTFast.cs would not recognize the file format. This was causing problems for me since I was getting GLTFs from a CDN that auto-generated the GLTFs with things like texture version info in the query strings. I fixed this by adding a set of "contains" checks after the "ends with" checks in GLTFast.cs. I added them at the end so that they will only be used if it does not end with a file format since I suspect that this will still be far more common.

### CODE REVIEW

File: Runtime/Scripts/GltFast.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿// Copyright 2020-2021 Andreas Atteneder
```
Comment (atteneder) [inline Runtime/Scripts/GltFast.cs:1]: Please avoid/remove non-changes like this. They screw up the blame-history of a document.

File: Runtime/Scripts/GltFast.cs [hunk 2]
```diff
@@ -2195,10 +2195,16 @@ async Task
 
         ImageFormat GetImageFormatFromPath(string path) {
             if(path.EndsWith(".png",StringComparison.OrdinalIgnoreCase)) return ImageFormat.PNG;
-            if(path.EndsWith(".jpg",StringComparison.OrdinalIgnoreCase)
```
Comment (atteneder) [inline Runtime/Scripts/GltFast.cs:2198]: Again, a white-space only change. Please remove them.

File: Runtime/Scripts/GltFast.cs [hunk 3]
```diff
@@ -2195,10 +2195,16 @@ async Task
...
+            if(path.EndsWith(".jpg",StringComparison.OrdinalIgnoreCase) 
                 || path.EndsWith(".jpeg",StringComparison.OrdinalIgnoreCase)) return ImageFormat.Jpeg;
             if(path.EndsWith(".ktx",StringComparison.OrdinalIgnoreCase)
                 || path.EndsWith(".ktx2",StringComparison.OrdinalIgnoreCase)) return ImageFormat.KTX;
+            if(path.IndexOf(".png", StringComparison.OrdinalIgnoreCase) >= 0) return ImageFormat.PNG;
```
Comment (atteneder) [inline Runtime/Scripts/GltFast.cs:2202]: This is somewhat redundant to the conditions above and has a potential of creating false positives.
A better approach would be to search for the last occurance of `.`, assume it separates the extension and look at the 3-4 following chars.
Have a look at `UriHelper.IsGltfBinary` for reference. It does the same for .gltf/.glb

**Review by atteneder — CHANGES_REQUESTED** [review]

File: Runtime/Scripts/GltFast.cs (hunk 1 above)
Comment (JonathanB-Vobling) [reply Runtime/Scripts/GltFast.cs:1]: Will do! My bad.

File: Runtime/Scripts/GltFast.cs (hunk 2 above)
Comment (JonathanB-Vobling) [reply Runtime/Scripts/GltFast.cs:2198]: Same as above! My bad.

File: Runtime/Scripts/GltFast.cs (hunk 3 above)
Comment (JonathanB-Vobling) [reply Runtime/Scripts/GltFast.cs:2202]: I thought of something like that but query strings can contain periods so that would still not resolve the issue.
The redundancy is another matter and will fix. Was a perhaps misguided attempt to have some holistic performance, since the majority of gltfs won't have query strings.

Comment (JonathanB-Vobling) [reply Runtime/Scripts/GltFast.cs:2202]: Since this is a specific fix to handle URL queries we could look from first "?" and forward to "." to find the file ending and compare that if you are worried about false positives for filenames or query strings with .png in them.

Comment (atteneder) [reply Runtime/Scripts/GltFast.cs]: That sounds good to me!

**Review by atteneder — APPROVED** [review]

**Comment by atteneder** [comment]
This looks good.

Usually I should have asked you in the first response to provide some unit test for this method (to make sure we don't break anything and don't miss corner cases), but I looked at it and decided this also needs to be restructured/refactored outside of the `GLTFast` class. I'm gonna merge now and do the refactor myself.

Thank you very much!

**Comment by atteneder** [comment]
@JonathanB-Vobling 

fyi: unit tests would've been useful indeed: I wrote some and revealed cases that led to crashes!

See my [commit](https://github.com/atteneder/glTFast/commit/3c4712383cac7eb53ccd5e36754ddfd7361973fb) to see what I changed.

Anyways, should be fine now. thanks
